core/flow/rule_manager.go

```diff
@@ -28,6 +29,7 @@ var (
 	tcGenFuncMap = make(map[trafficControllerGenKey]TrafficControllerGenFunc)
 	tcMap        = make(TrafficControllerMap)
 	tcMux        = new(sync.RWMutex)
+	currentRules = make([]*Rule, 0)
```

Comment:
Better to protect it by a lock to avoid concurrent read and write. And should we consider the `Rule` order of different resources? How about a map?

Reply:
Locks are used in onRuleUpdate func,don't have to think about different resources, because want to cache global rules.

Reply:
`LoadRules` reads `currentRules ` without lock.

Reply:
updated